libminifi/test/Utils.h

```diff
@@ -135,3 +142,37 @@ struct FlowFileQueueTestAccessor {
   FIELD_ACCESSOR(load_task_);
   FIELD_ACCESSOR(queue_);
 };
+
+bool sendMessagesViaSSL(const std::vector<std::string_view>& contents, uint64_t port, const std::string& ca_cert_path, const std::optional<minifi::utils::net::SslData>& ssl_data = std::nullopt) {
+  asio::ssl::context ctx(asio::ssl::context::sslv23);
+  ctx.load_verify_file(ca_cert_path);
+  if (ssl_data) {
+    ctx.set_verify_mode(asio::ssl::verify_peer);
+    ctx.use_certificate_file(ssl_data->cert_loc, asio::ssl::context::pem);
+    ctx.use_private_key_file(ssl_data->key_loc, asio::ssl::context::pem);
```

Comment:
Do we also need to set the password callback?

Reply:
Added in f3fb107052af05c62635ba3d4a9c1906c714d6fe